# C1_성민지_Python을 이용한 실습_220804

---



In [25]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

### Python을 이용한 기술통계 실습 (210쪽)
- 자동차 연비 Data Set에서 기술통계치 구하기

In [3]:
dat = pd.read_csv('/home/piai/minji/dataset/mycars.csv', encoding = 'euc-kr')

print(dat.shape)
dat.head()

(234, 11)


manufacturer model  displacement  year  cylinder automatic driving  mpg  \
0         audi    a4           1.8  1999         4      auto       f   18   
1         audi    a4           1.8  1999         4    manual       f   21   
2         audi    a4           2.0  2008         4    manual       f   20   
3         audi    a4           2.0  2008         4      auto       f   21   
4         audi    a4           2.8  1999         6      auto       f   16   

   highway_mileage fuel    class  
0               29    p  compact  
1               29    p  compact  
2               31    p  compact  
3               30    p  compact  
4               26    p  compact

In [5]:
dat.groupby('automatic').describe().round(3)

displacement     year  cylinder      mpg  highway_mileage
count       234.000   234.00   234.000  234.000          234.000
mean          3.472  2003.50     5.889   16.859           23.440
std           1.292     4.51     1.612    4.256            5.955
min           1.600  1999.00     4.000    9.000           12.000
25%           2.400  1999.00     4.000   14.000           18.000
50%           3.300  2003.50     6.000   17.000           24.000
75%           4.600  2008.00     8.000   19.000           27.000
max           7.000  2008.00     8.000   35.000           44.000

- 전체 데이터 개수는 234개이다.
- 년도는 1999년부터 2008년까지 분포되어 있는 것으로 확인된다.
- 차에 배치된 실린더의 개수는 일반적으로 5-6개이다.

### Python을 이용한 기술통계 연습 (212쪽)

In [7]:
dat = pd.read_csv('/home/piai/minji/dataset/mycars.csv', encoding = 'euc-kr')

print(dat.shape)
dat.head()

(234, 11)


manufacturer model  displacement  year  cylinder automatic driving  mpg  \
0         audi    a4           1.8  1999         4      auto       f   18   
1         audi    a4           1.8  1999         4    manual       f   21   
2         audi    a4           2.0  2008         4    manual       f   20   
3         audi    a4           2.0  2008         4      auto       f   21   
4         audi    a4           2.8  1999         6      auto       f   16   

   highway_mileage fuel    class  
0               29    p  compact  
1               29    p  compact  
2               31    p  compact  
3               30    p  compact  
4               26    p  compact

In [10]:
dat1 = dat[['automatic', 'model']]

print(dat1.shape)
dat1.head()

(234, 2)


automatic model
0      auto    a4
1    manual    a4
2    manual    a4
3      auto    a4
4      auto    a4

In [16]:
# 'automatic' 변수의 범주별 개수
count_auto = dat1['automatic'].value_counts().sort_index()


# 누적개수, 확률, 누적확률
cumcnt_auto = np.cumsum(count_auto)
percent_auto = count_auto / sum(count_auto) * 100
cumpct_auto = np.cumsum(percent_auto)

In [17]:
count_data_auto = pd.DataFrame({'Count' : count_auto, 'CumCnt' : cumcnt_auto,
                                'Percent' : percent_auto, 'CumPct' : cumpct_auto})

In [18]:
count_data_auto

Count  CumCnt    Percent      CumPct
auto      157     157  67.094017   67.094017
manual     77     234  32.905983  100.000000

**구동방식을 기준으로 데이터를 구분**
- 자동 157개와 수동 77개로 이루어져 있다는 것을 확인할 수 있다.
- manual의 CumCnt를 통해 전체 데이터의 개수가 234개인 것을 확인할 수 있다.
- auto는 전체 데이터의 약 67%를 차지한다.

In [19]:
# 'model' 변수의 범주별 개수
count_model = dat1['model'].value_counts().sort_index()


# 누적개수, 확률, 누적확률
cumcnt_model = np.cumsum(count_model)
percent_model = count_model / sum(count_model) * 100
cumpct_model = np.cumsum(percent_model)

In [20]:
count_data_model = pd.DataFrame({'Count' : count_model, 'CumCnt' : cumcnt_model,
                                'Percent' : percent_model, 'CumPct' : cumpct_model})

In [21]:
count_data_model

Count  CumCnt   Percent      CumPct
4runner 4wd                 6       6  2.564103    2.564103
a4                          7      13  2.991453    5.555556
a4 quattro                  8      21  3.418803    8.974359
a6 quattro                  3      24  1.282051   10.256410
altima                      6      30  2.564103   12.820513
c1500 suburban 2wd          5      35  2.136752   14.957265
camry                       7      42  2.991453   17.948718
camry solara                7      49  2.991453   20.940171
caravan 2wd                11      60  4.700855   25.641026
civic                       9      69  3.846154   29.487179
corolla                     5      74  2.136752   31.623932
corvette                    5      79  2.136752   33.760684
dakota pickup 4wd           9      88  3.846154   37.606838
durango 4wd                 7      95  2.991453   40.598291
expedition 2wd              3      98  1.282051   41.880342
explorer 4wd                6     104  2.564103   44.444444
f150 pickup 4wd             7     111  2.991453   47.435897
forester awd                6     117  2.564103   50.000000
grand cherokee 4wd          8     125  3.418803   53.418803
grand prix                  5     130  2.136752   55.555556
gti                         5     135  2.136752   57.692308
impreza awd                 8     143  3.418803   61.111111
jetta                       9     152  3.846154   64.957265
k1500 tahoe 4wd             4     156  1.709402   66.666667
land cruiser wagon 4wd      2     158  0.854701   67.521368
malibu                      5     163  2.136752   69.658120
maxima                      3     166  1.282051   70.940171
mountaineer 4wd             4     170  1.709402   72.649573
mustang                     9     179  3.846154   76.495726
navigator 2wd               3     182  1.282051   77.777778
new beetle                  6     188  2.564103   80.341880
passat                      7     195  2.991453   83.333333
pathfinder 4wd              4     199  1.709402   85.042735
ram 1500 pickup 4wd        10     209  4.273504   89.316239
range rover                 4     213  1.709402   91.025641
sonata                      7     220  2.991453   94.017094
tiburon                     7     227  2.991453   97.008547
toyota tacoma 4wd           7     234  2.991453  100.000000

**모델을 기준으로 데이터를 구분**
- caravan 2wd과 ram 1500 pickup 4wd이 여러 자동차 모델 중 가장 많이 분포되어 있는 것으로 확인된다.

### Python을 이용한 확률 계산 실습 (226쪽)

In [28]:
mu = 175
sigma = 5
x = 180

prob = stats.norm.cdf(x, mu, sigma)
print("나의 키가 {0}cm일 경우 내가 위치할 % 범위 : {1:.1f}%".format(x, (1-prob) * 100))

나의 키가 180cm일 경우 내가 위치할 % 범위 : 15.9%


### Python을 이용한 확률 계산 실습 (228쪽)

In [ ]:
mu = 12
sigma = 3
x1 = 15

prob1 = stats.norm.cdf(x1, mu, sigma)
print{"{0}km 이상 {1}km 이하를 달릴 확률은 {2:.1f}%에 해당함".format(mu, x1, (prob1 - 0.5) * 100)}

In [29]:
prob1 = stats.norm.cdf(15, 12, 3) - stats.norm.cdf(9, 12, 3)
print("{0}km 이상 {1}km 이하를 달릴 확률은 ? ".format(12, 15))

### Python을 이용한 확률 계산 실습 (236쪽)

In [30]:
t = 2.0
df = 100

prob = stats.t.cdf(t, df)
print("P(T <= F) : {:.3f}".format(prob))

P(T <= F) : 0.976


자유도가 100일 때, 2.0에 대한 검정통계량 T의 누적확률은 0.976이다.

### Python을 이용한 확률 계산 실습 (239쪽)

In [32]:
chisq = 3
df = 8

prob = stats.chi2.cdf(chisq, df)
print("P(X <= {0}) : {1:.4f}".format(chisq, prob))

P(X <= 3) : 0.0656


자유도가 8일 때, 3에 대한 검정통계량 X^2의 누적확률은 0.0656이다.

### Python을 이용한 확률분포 그리기, 확률 계산 실습 (243쪽)

In [34]:
f = 1.5
dfnum = 10
dfden = 100

prob = stats.f.cdf(x = f, dfn = dfnum, dfd = dfden)
print("P(X <= F): {:.3f}".format(prob))

P(X <= F): 0.850


자유도가 각각 10과 100일 때, F = 1.5에 대한 검정통계량 F의 누적확률은 0.850이다.

### Python을 이용한 확률분포 그리기, 확률 계산 실습 (250쪽)

In [36]:
n = 5
for i in range(n+1) :
    prob = stats.binom.pmf(k = i, n = n, p = 0.3)
    print("P(X={0}) = {1:.3f}".format(i, prob))

P(X=0) = 0.168
P(X=1) = 0.360
P(X=2) = 0.309
P(X=3) = 0.132
P(X=4) = 0.028
P(X=5) = 0.002


5대의 차량을 임의로 선택했을 때, 불량대수가 각각 0, 1, 2, 3, 4, 5일 확률은 각각 0.168, 0.360, 0.309, 0.132, 0.028, 0.002이다.